### ISSUE TITLE
Change Event Log Check previous days to more specific time period

### ISSUE THREAD

**Issue by hendymen** [amidaware/tacticalrmm#2090]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
I'd like to have the option to change the "How many previous days to search (Enter 0 for the entire log)" of an new Event Log Check to a smaller amount than 1 day. We want to check if an Event was logged in the last 5 minutes and then restart a certain service (by an attached task). We're experiencing issues with the check checking for a whole day. Because there can me multiple events during the day.

Or we would like it to only respond on new event log entries.

**Additional context**
Please see this picture

<img width="558" alt="image" src="https://github.com/user-attachments/assets/4a63ffee-5c18-4bd5-8f74-0f74acad4a57">

**Comment by P6g9YHK6** [comment]
i had the same kind of issue yesterday i could not get a "on failure" tasks to only run once for a bluescreen issue
what is missing is a on failure **once** tag when setting the tasks
![image](https://github.com/user-attachments/assets/3a8435a9-b4f4-47e6-b059-c28990499bf3)

**Comment by hendymen** [comment]
> i had the same kind of issue yesterday i could not get a "on failure" tasks to only run once for a bluescreen issue what is missing is a on failure **once** radio button when setting the tasks 1 run on each failure 2 run on the 1st failure only ![image](https://private-user-images.githubusercontent.com/17877371/392012134-3a8435a9-b4f4-47e6-b059-c28990499bf3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyMzc4MjMsIm5iZiI6MTczMzIzNzUyMywicGF0aCI6Ii8xNzg3NzM3MS8zOTIwMTIxMzQtM2E4NDM1YTktYjRmNC00N2U2LWIwNTktYzI4OTkwNDk5YmYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAzVDE0NTIwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3YmJmYjA2OWI5OTNlNjdjNjY1YWUzMWU0MDRiMzdlYzYxMGUwMWZkZTAyZmI2MzZmOTI5ZTk3NjM4YWQyZGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.B_mHTIMmcogIBHyF_lzUrMOrMJI9_Y9z5wKmXCqaPls)

Yeah great idea, so in fact, your task can still be "On Check Failure" but the check itself should have the option to only mark itself as failed if a new event was logged (not checking the history previously the past event)!

**Comment by hendymen** [comment]
Right now I am in a loop..

The check checks if the event was logged for the past day, it was logged, so it continues to the task, however the task restarts a service. However the restart of this service triggers a new event log.. And so on and so on.

But I can't use the check service, because the service keeps running despite the program fails, so it's not just a service check...

**Comment by P6g9YHK6** [comment]
for now you should just script it then you could easily deal with these "inconvenience"
